hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java

```diff
@@ -94,6 +95,10 @@ public OmSnapshot load(@Nonnull String snapshotTableKey)
         // see if the snapshot exists
         snapshotInfo = getSnapshotInfo(snapshotTableKey);
 
+        boolean isSnapshotInCache =
+            ozoneManager.getMetadataManager().getSnapshotInfoTable()
+                .getCacheValue(new CacheKey<>(snapshotTableKey)) != null;
```

Comment:
Is this accounting for the case where the snapshot entry is a tombstone in cache (`CacheValue` is `Optional.absent()` like [here](https://github.com/apache/ozone/blob/ddbe71d3f90ae14ecf29f51c2265f1f9c3171668/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyDeleteRequest.java#L158-L162)), which could happen when the snapshot eventually deleted from the snapshotInfo table? Doesn't look to be the case to me?

Therefore, IMO in addition to checking null, we would need to check whether the value is `Optional.absent()`. If it is `Optional.absent()`, `isSnapshotInCache` should still be `false` as we won't want to wait for a checkpoint dir that is deleted. Or we would have to put checks before this logic is even reached.

Reply:
nvm. I checked that this case is also handled by the `null` check.

Reply:
Thanks for finding this. Didn't get this, how is it covered by null check.

Reply:
I ran a UT and `getCacheValue` also returns `null` in this case

<img width="1535" alt="cachevalue" src="https://user-images.githubusercontent.com/50227127/220413948-aa5f3b38-670e-43bb-b714-2e0efda14415.png">

Reply:
Uh oops. My bad. The value inside the object is `null`. But the object itself is not `null`.

@sadanand48 Would you like to post an addendum to this jira?

Reply:
Sure I can raise another PR with this change, Is a revert required here or should I only include this particular check in new PR?

Reply:
@sadanand48 No revert needed here. The addendum only needs the new changes.